Generated BUILD.bazel file is missing `_windows_arm64.go` from srcs

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version of gazelle are you using?

v0.30.0

### What version of rules_go are you using?

v0.39.0

### What version of Bazel are you using?

```
Build label: 6.1.1
Build target: bazel-out/darwin_arm64-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Wed Mar 15 15:45:56 2023 (1678895156)
Build timestamp: 1678895156
Build timestamp as int: 1678895156
```

### Does this issue reproduce with the latest releases of all the above?

Yes


### What operating system and processor architecture are you using?

Darwin and linux hosts


### What did you do?

built go code with an golang.org/x/sys dependency with `GOOS=windows GOARCH=arm64` 

### What did you expect to see?

Binary built successfully

### What did you see instead?

```
ERROR: /private/var/tmp/_bazel_connor/072704f4f617c1b547cf8cf8993087b4/external/org_golang_x_sys/windows/BUILD.bazel:3:11: GoCompilePkg external/org_golang_x_sys/windows/windows.a failed: (Exit 1): builder failed: error executing command (from target @org_golang_x_sys//windows:windows) bazel-out/darwin_arm64-opt-exec-2B5CBBC6-ST-e5dc23cab020/bin/external/go_sdk/builder_reset/builder compilepkg -sdk external/go_sdk -installsuffix windows_arm64 -src ... (remaining 71 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
external/org_golang_x_sys/windows/types_windows.go:2198:24: undefined: JOBOBJECT_BASIC_LIMIT_INFORMATION
external/org_golang_x_sys/windows/zsyscall_windows.go:4138:38: undefined: WSAData
external/org_golang_x_sys/windows/zsyscall_windows.go:4214:51: undefined: Servent
external/org_golang_x_sys/windows/zsyscall_windows.go:4228:50: undefined: Servent
external/org_golang_x_sys/windows/zsyscall_windows.go:4230:8: undefined: Servent
compilepkg: error running subcommand external/go_sdk/pkg/tool/darwin_arm64/compile: exit status 2
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 2.410s, Critical Path: 0.23s
INFO: 2 processes: 2 internal.
FAILED: Build did NOT complete successfully
```

--- a/language/go/fileinfo.go
+++ b/language/go/fileinfo.go
@@ -621,7 +621,8 @@ func rulesGoSupportsPlatform(v version.Version, p rule.Platform) bool {
 			p.OS == "linux" && p.Arch == "riscv64" ||
 			p.OS == "netbsd" && p.Arch == "arm64" ||
 			p.OS == "openbsd" && p.Arch == "arm64" ||
-			p.OS == "windows" && p.Arch == "arm") {
+			p.OS == "windows" && p.Arch == "arm" ||
+			p.OS == "windows" && p.Arch == "arm64") {
 		return false
 	}
 	return true

--- a/rule/platform.go
+++ b/rule/platform.go
@@ -94,6 +94,7 @@ var KnownPlatforms = []Platform{
 	{"windows", "386"},
 	{"windows", "amd64"},
 	{"windows", "arm"},
+	{"windows", "arm64"},
 }
 
 var OSAliases = map[string][]string{

